incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -411,13 +416,17 @@ func (r *HierarchyConfigReconciler) syncConditions(log logr.Logger, inst *api.Hi
 
 // syncCritConditions enqueues the children of a namespace if the existing critical conditions in the
 // namespace are gone or critical conditions are newly found.
-func (r *HierarchyConfigReconciler) syncCritConditions(log logr.Logger, ns *forest.Namespace, hadCrit bool) {
+func (r *HierarchyConfigReconciler) syncCritConditions(log logr.Logger, ns *forest.Namespace, deletingCRD, hadCrit bool) {
 	// If we're in a cycle, determine that now
 	if cycle := ns.CycleNames(); cycle != nil {
 		msg := fmt.Sprintf("Namespace is a member of the cycle: %s", strings.Join(cycle, " <- "))
 		ns.SetLocalCondition(api.CritCycle, msg)
 	}
 
+	if deletingCRD {
+		ns.SetLocalCondition(api.CritDeletingCRD, "The HierarchyConfiguration CRD is being deleted; all syncing is disabled.")
```

Comment:
Is `CritDeletingCRD` condition very short-lived? Considering it will be gone when the CRD and CR are deleted.

Reply:
Yup. Its _main_ purpose it to stop the propagated objects from being deleted.